resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -780,10 +780,19 @@ private[spark] class Client(
       props.store(writer, "Spark configuration.")
       writer.flush()
       confStream.closeEntry()
+      confArchive
+    } catch {
+      case e: IOException =>
+        logError(s"IOException occurred while writing to confArchive $confArchive", e)
+        throw e
```

Comment:
Why do we need to log error when the exception is thrown?

Reply:
@HyukjinKwon because as explained in the jira https://issues.apache.org/jira/browse/SPARK-26529, the message of some IOException like "No space left exception" does not contains the writing path information, so will delay the  locating of the root cause(which local disk, which files consuming large space)

Reply:
Then, we can just add a debug log somewhere. I don't think we're going to whitelist every possible exceptions and add some information here.

Reply:
I meant, I think I wouldn't list more exceptions when similar case happens. For instance, current code only handles with `IOException`. If other exceptions can be raised again, we should add it again. It sounds not quite reasonable.

So, I would rather do, for instance, `logDebug("Creating an archive with the config files for distribution at $confArchive.")` in the main code.

Reply:
OK, I Got it! maybe you are right for common uses, but in our case, my modification works well, for some online services like sparksql, debug has a time lag.